static/js/compose_state.js

```diff
@@ -45,6 +45,23 @@ export function focus_in_empty_compose() {
     return composing() && message_content() === "" && $("#compose-textarea").is(":focus");
 }
 
+export function focus_in_empty_compose_recipient_box() {
+    if (!composing()) {
+        return false;
+    }
+
+    const focused_element_id = document.activeElement.id;
+    switch (focused_element_id) {
+        case "private_message_recipient":
+            return private_message_recipient().length === 0;
+        case "stream_message_recipient_topic":
+        case "stream_message_recipient_stream":
+            return document.activeElement.value === "";
+    }
+
+    return false;
+}
```

Comment:
I think we still want to check that `message_content() === ""` as well?  

One way to do this might be to extract an `compose_open_and_empty` helper that both this function and `focus_in_empty_compose` would call, and then the differing logic would just be for where the focus is.

Though ... could we actually just be modifying `focus_in_empty_compose` itself to have additional cases, rather than adding a separate function?  I'm not sure why the `hotkey.js` logic would be different here than before.  In this proposal, `focus_in_empty_compose` would become:
```
if (!composing()) {
   return false;
}

if (message_content() !== "") {
    return false;
}

const focused_element_id = document.activeElement.id;
...
```

Reply:
Yeah, makes sense to do it in the same function.